java/src/jmri/jmrit/display/layoutEditor/BlockValueFile.java

```diff
@@ -71,48 +74,78 @@ public void readBlockValues() throws JDOMException, IOException {
                 if (blockvalues != null) {
                     // there are values defined, read and set block values if Block exists.
                     List<Element> blockList = blockvalues.getChildren("block");
-                    for (int i = 0; i < blockList.size(); i++) {
-                        if ((blockList.get(i)).getAttribute("systemname") == null) {
-                            log.warn("unexpected null in systemName "
-                                    + blockList.get(i) + " "
-                                    + blockList.get(i).getAttributes());
-                            break;
+                    // check if all powermanagers are turned on, if they are, we should expect
+                    // blocks with values to be occupied
+                    boolean allPoweredUp = true;
+                    for (PowerManager pm : jmri.InstanceManager.getList(PowerManager.class)) {
+                        try {
+                            if (pm.getPower() != jmri.PowerManager.ON) {
+                                allPoweredUp = false;
+                            }
+                        } catch (JmriException e) {
+                            allPoweredUp = false;
                         }
-                        String sysName = blockList.get(i).
-                                getAttribute("systemname").getValue();
-                        // get Block - ignore entry if block not found
-                        Block b = blockManager.getBySystemName(sysName);
-                        if (b != null) {
-                            // Block was found, set its value
-                            Object v = blockList.get(i).
-                                    getAttribute("value").getValue();
-                            if (blockList.get(i).getAttribute("valueClass") != null) {
-                                if (blockList.get(i).getAttribute("valueClass").getValue().equals("jmri.jmrit.roster.RosterEntry")) {
-                                    RosterEntry re = Roster.getDefault().getEntryForId(((String) v));
-                                    if (re != null) {
-                                        v = re;
+                    }
+                    List<String> passes = new ArrayList<>();
```

Comment:
No need to change it, this code is fine, but as a future style suggestion I think that an approach that uses
```
enum Pass { CHECK, SET}
```
instead of `String` values adds some clarity: 
- The reader knows exactly what values are possible, so there's no wondering if there are other possibilities
- It gives you a natural place for a comment on what each pass does
- Somebody working on the code later can't make an unnoticed typo on the value